crew

```diff
@@ -731,62 +760,77 @@ def remove (pkgName)
 
 end
 
-case @command
-when "help"
-  if @pkgName
-    help @pkgName
-  else
-    puts "Usage: crew help [command]"
-    help nil
-  end
-when "search"
-  if @pkgName
-    regexp_search @pkgName
-  else
-    list_packages
-  end
-when "whatprovides"
-  if @pkgName
-    whatprovides @pkgName
-  else
-    help "whatprovides"
+def build_command (args)
+  args["<name>"].each do |name|
+    @pkgName = name
+    search @pkgName
+    resolve_dependencies_and_build
   end
-when "download"
-  if @pkgName
+end
+
+def download_command (args)
+  args["<name>"].each do |name|
+    @pkgName = name
     search @pkgName
     download
+  end
+end
+
+def help_command (args)
+  if args["<command>"]
+    help args["<command>"]
   else
-    help "download"
+    puts "Usage: crew help [command]"
+    help nil
   end
-when "update"
-  update
-when "upgrade"
-  upgrade
-when "install"
-  if @pkgName
+end
+
+def install_command (args)
+  args["<name>"].each do |name|
+    @pkgName = name
     search @pkgName
     resolve_dependencies_and_install
-  else
-    help "install"
   end
-when "build"
-  if @pkgName
-    search @pkgName
-    resolve_dependencies_and_build
-  else
-    help "build"
+end
+
+def remove_command (args)
+  args["<name>"].each do |name|
+    remove name
   end
-when "remove"
-  if @pkgName
-    remove @pkgName
-  else
-    help "remove"
+end
+
+def search_command (args)
+  args["<name>"].each do |name|
+    regexp_search name
+  end.empty? and begin
+    list_packages
   end
-when nil
-  puts "Chromebrew, version 0.4.3"
-  puts "Usage: crew [command] [package]"
-  help nil
-else
-  puts "I have no idea how to do #{@command} :(".lightred
-  help nil
 end
+
+def update_command (args)
+  update
+end
+
+def upgrade_command (args)
+  args["<name>"].each do |name|
+    @pkgName = name
+    upgrade
+  end.empty? and begin
+    upgrade
+  end
+end
+
+def whatprovides_command (args)
+  args["<name>"].each do |name|
+    whatprovides name
+  end
+end
+
+def is_command (name)
+  return false if name =~ /^[-<]/
+  return true
+end
+
+command_name = args.find { |k, v| v && is_command(k) } [0]
+function = command_name + "_command"
+send(function, args)
```

Comment:
So what behavior you want?  It smells like you choose opposite whatever I answer.

Anyway, docopt consideres both are switches.